IdentityCore/tests/MSIDPkceTests.m

```diff
@@ -0,0 +1,62 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import <XCTest/XCTest.h>
+#import "MSIDPkce.h"
+
+NSUInteger const kCodeVerifierMinLength = 43;
+NSUInteger const kCodeVerifierMaxLength = 128;
+
+@interface MSIDPkceTests : XCTestCase
+
+@end
+
+@implementation MSIDPkceTests
+
+- (void)setUp {
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+- (void)testPkce_whenInit_shouldInitAllPropertiesToValidValue
+{
+    MSIDPkce *pkce = [MSIDPkce new];
+    XCTAssertTrue(pkce.codeVerifier.length >= kCodeVerifierMinLength);
```

Comment:
don't we have a predefined length that we're using? if yes, unit test should test it.

Reply:
This test is to make sure our final codeVerifier is within the limits of the standard, in case in future, if (big if in reality though) someone changes the implementation.